code/drasil-lang/lib/Drasil/Database/UID.hs

```diff
@@ -23,51 +23,42 @@ class HasUID c where
   -- | Provides a /unique/ id for internal Drasil use.
   uid :: Getter c UID
 
--- | A @UID@ is a 'unique identifier' for things that we will put into our database
--- of information. We use a newtype wrapper to make sure we are only using
--- 'UID's where desired.
+-- | A @UID@ is a 'unique identifier' for things that we will put into our
+-- database of information. We use a newtype wrapper to make sure we are only
+-- using 'UID's where desired.
 data UID = UID { _namespace :: [String], _baseName :: String }
-  deriving (Eq, Ord, Generic) --, ToJSONKey)
+  deriving (Eq, Ord, Generic)
 
 makeLenses ''UID
 
-fullName :: UID -> [String]
-fullName u = u ^. namespace ++ [u ^. baseName]
-
 instance ToJSON UID where
-  toJSON = toJSON . fullName
+  toJSON = toJSON . show
 
 instance ToJSONKey UID where
   toJSONKey = toJSONKeyText (pack . show)
 
 instance Show UID where
-  show = intercalate ":" . fullName
+  show u = intercalate ":" $ u ^. baseName : u ^. namespace
```

Comment:
Note that I inlined `fullName` here, but I also moved the "basename" to the front.